prow/gerrit/adapter/adapter.go

```diff
@@ -288,19 +288,33 @@ func (c *Controller) Sync() {
 		c.tracker.Update(latest)
 	}
 
+	// Identify projects without worker threads
+	id := func(instance, project string) string { return fmt.Sprintf("%s/%s", instance, project) }
+	needsWorker := map[string][]string{}
+	needsWorkerCount := 0
 	for instance, projects := range c.config().Gerrit.OrgReposConfig.AllRepos() {
 		for project := range projects {
-			id := fmt.Sprintf("%s/%s", instance, project)
-			if _, ok := c.projectsWithWorker[id]; ok {
+			if _, ok := c.projectsWithWorker[id(instance, project)]; ok {
 				// The worker thread is already up for this project, nothing needs
 				// to be done.
 				continue
 			}
-			c.projectsWithWorker[id] = true
-
-			// First time see this instance, spin up a worker thread for it
+			needsWorker[instance] = append(needsWorker[instance], project)
+			needsWorkerCount++
+		}
+	}
+	// First time seeing these projects, spin up worker threads for them.
+	newWorkerCount := 0
+	for instance, projects := range needsWorker {
+		for _, project := range projects {
+			c.projectsWithWorker[id(instance, project)] = true
 			logrus.WithFields(logrus.Fields{"instance": instance, "repo": project}).Info("Starting worker for project.")
-			go func(instance, project string) {
+			go func(instance, project string, workerCount int) {
+				// Stagger new worker threads across the loop period to reduce load on the Gerrit API and Git server.
+				napTime := (c.config().Gerrit.TickInterval.Duration / time.Duration(needsWorkerCount)) * time.Duration(workerCount)
```

Comment:
This line is a little hard to unpack mentally. I see that `needsWorkerCount` simply equals the total number of projects. How about just renaming it to `projectCount` ?

Here's my understanding: so we're dividing the TickInterval duration by the number of all projects. This part makes sense. And then we're simply asking each of these workers to wait an increasing number of seconds so that each of these workers will wake up one after the other to do their work (this is the `* time.Duration(workerCount)` part).

LGTM, but I think the first part of the calculation (TickInterval / needsWorkerCount) should probably be done outside this loop because this calculation is constant and there's no need to do it inside here. And maybe name it as `staggerIncrement` or `staggerAmount`?

Reply:
`needsWorkerCount` isn't just the number of projects, its the number of projects that don't have worker yet. That is what I mean to distinguish. 

Moving the `staggerIncrement` calculation outside the loop is a good idea, I've done that.